doc/source/io.rst

```diff
@@ -4809,6 +4809,30 @@ default ``Text`` type for string columns:
     Because of this, reading the database table back in does **not** generate
     a categorical.
 
+Datetime data types
+'''''''''''''''''''
+
+Using SQLAlchemy, :func:`~pandas.DataFrame.to_sql` is capable of writing
+datetime data that is timezone naive or timezone aware. However, the resulting
+data stored in the database ultimately depends on the corresponding,
+supported data type for datetime data.
```

Comment:
With "corresponding", you mean depending on the database system? 
If so, I think something like "this ultimately depends on the supported data type for datetime data of the database system being used" is clearer. 

I would also add that the following table lists some common systems (but that it can still be different for other database dialects)